ksqldb-functional-tests/src/test/resources/query-validation-tests/fk-join.json

```diff
@@ -3,92 +3,197 @@
     {
       "name": "Should fail on left non-key attribute for inner-join",
       "statements": [
-        "CREATE TABLE left_table (id1 BIGINT PRIMARY KEY, f1 BIGINT) WITH (kafka_topic='left_topic', format='JSON');",
```

Comment:
I did some more side cleanup to have a unified test and naming pattern.

Moved all negative tests to the top, and grouped them on "generally not supported", invalid join condition, invalid projection.

Reply:
Can we keep large refactors like this in separate PRs next time please? These test changes are really hard to review now -- I can't tell which changes are part of the refactor versus actual changes as part of this PR.

I think they're all just refactors? Hard to tell since tests were moved and renamed in the same change.

Reply:
This PR does not do any changes to test. (otherwise I would have pointed it out).